core/domain/voiceover_domain.py

```diff
@@ -104,101 +112,97 @@ def to_dict(self) -> EntityVoiceoverDict:
             given dict.
         """
         content_id_to_voiceovers = {}
         for content_id, voiceover_type_to_voiceover_dict in (
-                entity_voiceover_dict['voiceovers'].items()):
+                entity_voiceovers_dict['voiceovers'].items()):
             content_id_to_voiceovers[content_id] = {
-                'manual': state_domain.Voiceover.from_dict(
-                    voiceover_type_to_voiceover_dict['manual']),
-                'auto': state_domain.Voiceover.from_dict(
-                    voiceover_type_to_voiceover_dict['auto'])
+                feconf.VoiceoverType.MANUAL: (
+                    state_domain.Voiceover.from_dict(
+                        voiceover_type_to_voiceover_dict[
+                            feconf.VoiceoverType.MANUAL.value])),
+                feconf.VoiceoverType.AUTO: (
+                    state_domain.Voiceover.from_dict(
+                        voiceover_type_to_voiceover_dict[
+                            feconf.VoiceoverType.AUTO.value]))
             }
 
         return cls(
-            entity_voiceover_dict['entity_id'],
-            entity_voiceover_dict['entity_type'],
-            entity_voiceover_dict['entity_version'],
-            entity_voiceover_dict['language_accent_code'],
+            entity_voiceovers_dict['entity_id'],
+            entity_voiceovers_dict['entity_type'],
+            entity_voiceovers_dict['entity_version'],
+            entity_voiceovers_dict['language_accent_code'],
             content_id_to_voiceovers
         )
 
     def validate(self) -> None:
-        """Validates the EntityVoiceover object."""
+        """Validates the EntityVoiceovers object."""
         if not isinstance(self.entity_type, str):
             raise utils.ValidationError(
-                'entity_type must be a string, recieved %r' % self.entity_type)
+                'entity_type must be a string, received %s' % self.entity_type)
         if not isinstance(self.entity_id, str):
             raise utils.ValidationError(
-                'entity_id must be a string, recieved %r' % self.entity_id)
+                'entity_id must be a string, received %s' % self.entity_id)
         if not isinstance(self.entity_version, int):
             raise utils.ValidationError(
-                'entity_version must be an int, recieved %r' %
+                'entity_version must be an int, received %s' %
                 self.entity_version)
         if not isinstance(self.language_accent_code, str):
             raise utils.ValidationError(
-                'language_accent_code must be a string, recieved %r' %
+                'language_accent_code must be a string, received %s' %
                 self.language_accent_code)
-
+        if not bool(re.match(
+                feconf.LANGUAGE_ACCENT_CODE_REGEX, self.language_accent_code)):
+            raise utils.ValidationError(
+                'language_accent_code must follow the pattern language-accent'
```

Comment:
I'm not sure this will be understandable, maybe say something like "must be formatted as {{language code}}-{{accent code}}" or similar.

Reply:
Added, thank you